src/Parser.js

```diff
@@ -33,19 +36,41 @@ class Parser {
   }
 
   registerExtension(ext, parser) {
-    if (!ext.startsWith('.')) {
+    const type = toType(ext);
+
+    if (type === 'regexp' || isRegexp.test(ext)) {
+      ext = ext.toString();
+    } else if (!ext.startsWith('.')) {
       ext = '.' + ext;
     }
 
     this.extensions[ext] = parser;
   }
 
+  // find extension key
+  findExt(ext) {
+    const keys = Object.keys(this.extensions);
+
+    for (let i = keys.length - 1; i >= 0; i--) {
+      const key = keys[i];
+
+      if (isRegexp.test(key) && new RegExp(key, 'i').test(ext)) {
+        return key;
+      } else if (ext === key) {
+        key;
```

Comment:
`key;`  <- intentional?

Reply:
sorry forgot to remove test code 😔  I will fixed these issuses asap